File: 05_Arrays/src/Add_to_Array_Form_of_Integer989.java
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

public class Add_to_Array_Form_of_Integer989 {
    public static void main(String[] args) {
        System.out.println(addToArrayForm(new int[] {1,2,0,0}, 36));
        System.out.println(addToArrayForm(new int[] {2,7,4}, 181));
        System.out.println(addToArrayForm(new int[] {2,1,5}, 806));
    }
    public static List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> ansArr = new ArrayList<>();

        for (int i = 0; i < num.length; i++) {
            int sum = num[num.length-1-i]+k;
            ansArr.add(sum%10);
            k = sum/10;
        }

        while (k > 0) {
            ansArr.add(k%10);
            k = k/10;
        }

        Collections.reverse(ansArr);
        return ansArr;
    }
}

File: 05_Arrays/src/Add_to_Array-Form_of_Integer989.js
/**
 * @param {number[]} num
 * @param {number} k
 * @return {number[]}
 */
const addToArrayForm = function (num, k) {
    let ansArr = [];

    for (let i = num.length-1; i >= 0; i--) {
        let sum = num[i]+k;
        ansArr.push(sum%10);
        k = Math.floor(sum/10);
    }

    while (k > 0) {
        ansArr.push(k%10);
        k = Math.floor(k/10);
    }

    ansArr.reverse();

    return ansArr;
};

console.log(addToArrayForm([1,2,0,0],34))
console.log(addToArrayForm([2,7,4],181))
console.log(addToArrayForm([2,1,5],806))
console.log(addToArrayForm([1,2,6,3,0,7,1,7,1,9,7,5,6,6,4,4,0,0,6,3],516))
